src/utils/dif_upload.rs

```diff
@@ -286,6 +267,40 @@ impl fmt::Debug for DifMatch<'_> {
     }
 }
 
+impl AsRef<[u8]> for DifMatch<'_> {
+    fn as_ref(&self) -> &[u8] {
+        self.data()
+    }
+}
+
+trait ToAssemble {
+    fn to_assemble(&self, with_debug_id: bool) -> (Digest, ChunkedDifRequest<'_>);
+}
+
+impl ToAssemble for ChunkedObject<DifMatch<'_>> {
+    /// Creates a tuple which can be collected into a `ChunkedDifRequest`.
+    // Some(...) for debug_id can only be done if the ChunkedUploadCapability::Pdbs is
+    // present, which is kind of a protocol bug.  Not supplying it means more recent
+    // sentry-cli versions keep working with ancient versions of sentry by not
+    // triggering this protocol bug in most common situations.
+    // See: https://github.com/getsentry/sentry-cli/issues/980
+    // See: https://github.com/getsentry/sentry-cli/issues/1056
+    fn to_assemble(&self, with_debug_id: bool) -> (Digest, ChunkedDifRequest<'_>) {
+        (
+            self.checksum(),
+            ChunkedDifRequest {
+                name: self.object().file_name(),
+                debug_id: if with_debug_id {
+                    self.object().debug_id
+                } else {
+                    None
+                },
```

Comment:
This could be written using `filter`, not sure which is more elegant.
```suggestion
                debug_id: self.object().debug_id.filter(|_| with_debug_id),
```

Reply:
Let's leave it as is for now, I think it is out of scope for this PR to make this change. I just want to make the changes needed to migrate ChunkedDifMatch to a generic struct here